tdp/core/runner/deployment_plan.py

```diff
@@ -51,6 +53,49 @@ def from_dag(
     def from_operations(operations):
         deployment_args = dict(
             targets=[operation.name for operation in operations],
-            using_dag=False,
+            deployment_type=DeploymentTypeEnum.OPERATIONS,
         )
         return DeploymentPlan(operations, deployment_args)
+
+    @staticmethod
+    def from_failed_deployment(dag, deployment_log_to_resume):
+
+        if deployment_log_to_resume.deployment_type == DeploymentTypeEnum.DAG:
+            deployment_plan_to_resume = DeploymentPlan.from_dag(
+                dag,
+                deployment_log_to_resume.targets,
+                deployment_log_to_resume.sources,
+                deployment_log_to_resume.filter_expression,
+            )
+        elif deployment_log_to_resume.deployment_type == DeploymentTypeEnum.OPERATIONS:
+            raise Exception(
+                f"Resuming from an Operations deployment is not yet supported"
+            )
+        elif deployment_log_to_resume.deployment_type == DeploymentTypeEnum.RESUME:
+            raise Exception(f"Resuming from an Resume deployment is not yet supported")
+
+        if deployment_log_to_resume.state == StateEnum.SUCCESS:
+            raise Exception(f"Nothing to resume, deployment #{id} was successful")
```

Comment:
This exception should be a named exception like `NothingToResumeError` because this logic is inside a library. The 2 previous exceptions are temporary so that's fine with me.